### PR TITLE
Enable python 3.5.0-3.5.2

### PR DESCRIPTION
**Pull request by max-sixty**
- [x] Closes some of https://github.com/pydata/xarray/issues/2830

### CODE REVIEW

**Comment by max-sixty** [comment]
TBC, I don't think this would allow `3.5.2` to work from PyPI even if we did a point release - `3.5.2` would still run `0.12.0`

**Comment by max-sixty** [comment]
From #2830:

> Probably the simplest thing to do would be to guard the import in a typing.TYPE_CHECKING block

Unfortunately that's only [supported from 3.5.2](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING)! 

I'll add a version check. I think there are only two places, and it won't last long

**Comment by max-sixty** [comment]
I added a fix; it may have some mistakes which I'll correct

Is the approach OK? It's not pretty

File: xarray/coding/cftime_offsets.py [hunk 1]
```diff
@@ -41,15 +41,20 @@
...
-from typing import ClassVar, Optional
 
 import numpy as np
 
 from .cftimeindex import CFTimeIndex, _parse_iso8601_with_reso
+from .core.pycompat import USE_TYPING
 from .times import format_cftime_datetime
 
+if USE_TYPING:
+    if typing.TYPE_CHECKING:
```
Comment (shoyer) [inline xarray/coding/cftime_offsets.py]: Can mypy interpret this if we it this on one line? e.g., `if USE_TYPING and typing.TYPE_CHECKING` ?

**Comment by max-sixty** [comment]
Green! Good idea re the single line, thanks @shoyer

**Comment by max-sixty** [comment]
Please could someone else take a final check and merge? @shoyer reviewed but not since some changes. Thanks!

**Comment by shoyer** [comment]
This looks good to me. I tested it locally and it works with mypy.

I pushed a small change to put a single `TYPE_CHECKING` variable in pycompat, so we can simply the check that we put in each file.

**Comment by max-sixty** [comment]
> I pushed a small change to put a single TYPE_CHECKING variable in pycompat, so we can simply the check that we put in each file.

Much nicer!

**Pull request merged**